### PR TITLE
Build a docker image to run the evaluation in an isolated environment

### PR DESCRIPTION
**Pull request by Munsio**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+# Install Java
+RUN wget https://download.oracle.com/java/22/latest/jdk-22_linux-x64_bin.tar.gz && \
+	tar -xf jdk-22_linux-x64_bin.tar.gz -C ~/.eval-dev-quality/ && \
+	rm jdk-22_linux-x64_bin.tar.gz
+ENV JAVA_HOME="/root/.eval-dev-quality/jdk-22.0.1"
+ENV PATH="${PATH}:${JAVA_HOME}/bin"
+
+# Install Go
+RUN wget https://go.dev/dl/go1.22.4.linux-amd64.tar.gz && \
```
Comment (zimmski) [inline Dockerfile:26]: Please use 1:1 the same versions we use locally right now. So Go 1.21.5 here (for the others, i did not check) reason is that that is what we use in the CI, and we don't have "works here" problems.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,37 @@
+FROM ubuntu:noble
+RUN apt-get update && apt-get install -y ca-certificates wget unzip git make && update-ca-certificates
+
+RUN mkdir -p ~/.eval-dev-quality
```
Comment (zimmski) [inline Dockerfile:4]: Please use an "eval" user here. So we do not run as root. that helps with lots of isolation problems

File: README.md [hunk 3]
```diff
@@ -135,6 +135,23 @@ Total coverage 100.000000%
...
+### Build the image
+```bash
+docker build . -t eval-dev-quality
+```
+
+### Run the evaluation
+The following command will run the model `symflower/symbolic-execution` and stores the the results of the run inside the local directory `evaluation`.
+```bash
+docker run -v ./:/home/ubuntu/evaluation --user $(id -u):$(id -g) eval-dev-quality:latest eval-dev-quality evaluate --model symflower/symbolic-execution --result-path /home/ubuntu/evaluation/%datetime%
+```
```
Comment (bauersimon) [inline README.md:153]: Maybe add that one can also pull the image right away?

File: .github/workflows/docker.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+          tags: |
+            type=raw,value={{ branch }}
+            type=raw,value=latest,enable={{ is_default_branch }}
+      - name: Build and push
+        uses: docker/build-push-action@v5
+        with:
+          context: ./
+          platforms: linux/amd64
+          tags: ${{ steps.meta.outputs.tags }}
+          push: true
```
Comment (bauersimon) [inline .github/workflows/docker.yml:43]: We could already add the commit hash here that we need for #207, right?

File: README.md [hunk 5]
```diff
@@ -152,6 +152,10 @@ The following command will run the model `symflower/symbolic-execution` and stor
 docker run -v ./:/home/ubuntu/evaluation --user $(id -u):$(id -g) eval-dev-quality:latest eval-dev-quality evaluate --model symflower/symbolic-execution --result-path /home/ubuntu/evaluation/%datetime%
 ```
 
+# Kubernetes
+
+Please check the [Kubernetes](./docs/kubernetes/README.md) documentation.
+
```
Comment (bauersimon) [inline README.md:158]: Looks pretty but TBH I have no idea what is going on ^^

File: scripts/docker-evaluation/docker.go [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+	"openrouter/togethercomputer/stripedhyena-nous-7b",
+	"openrouter/undi95/toppy-m-7b",
+	"openrouter/xwin-lm/xwin-lm-70b",
+}
+
+func main() {
+
+	for _, model := range models {
+
+		split := strings.Split(model, "/")
```
Comment (bauersimon) [inline scripts/docker-evaluation/docker.go:96]: plz do

`modelIdentifier = strings.Split(model, provider.ProviderModelSeparator)[2]`

**Review by bauersimon — CHANGES_REQUESTED** [review]

File: .github/workflows/docker.yml (hunk 4 above)
Comment (Munsio) [reply .github/workflows/docker.yml:43]: Will do that with #207

**Review by bauersimon — APPROVED** [review]

**Pull request merged**